grove/plugin/plugin.go

```diff
@@ -91,6 +92,12 @@ type BeforeRespondData struct {
 	Context   *interface{}
 }
 
+type BeforeCacheLookUpData struct {
```

Comment:
Any objection to putting the overrideFunc in here? So it'd be  `type BeforeCacheLookUpData struct { OverrideCacheKey func(string) ...` and `d.OverrideCacheKey("foo")` ? IMO it'd be cleaner and simpler, to avoid adding the extra parameter to everything.